### PR TITLE
[WIP] Integration of baremetal ipi test in release controller for branch 4.5

### PR DESCRIPTION
**Pull request by andfasano**
This PR aims to integrate the baremetal ipi test (see the step registry [1] for more details about it) in the release controller, so that it will appear in in https://openshift-release.svc.ci.openshift.org/. 

[1] https://github.com/openshift/release/tree/master/ci-operator/step-registry/baremetalds/e2e/baremetalds-e2e-workflow.yaml

### CODE REVIEW

**Comment by smarterclayton** [comment]
Your rehearse failed with:

```
time="2020-03-25T11:00:02Z" level=warning msg="Failed to inline ci-operator-config into rehearsal periodic job" error="ci-operator config file --.yaml was not found" pr=7905 target-job=release-openshift-ocp-installer-e2e-metal-ipi-4.5
```

File: ci-operator/jobs/openshift/release/openshift-release-release-4.5-periodics.yaml [hunk 1]
```diff
@@ -6077,3 +6077,75 @@ periodics:
...
         secretName: ci-pull-credentials
+- agent: kubernetes
+  cluster: api.ci
+  decorate: true
+  interval: 48h
+  labels:
+    ci.openshift.io/release-type: informing
+    job-release: "4.5"
+    pj-rehearse.openshift.io/can-be-rehearsed: "true"
+  name: release-openshift-ocp-installer-e2e-metal-ipi-4.5
```
Comment (stbenjam) [inline ci-operator/jobs/openshift/release/openshift-release-release-4.5-periodics.yaml]: So this job does look a little different than for example `release-openshift-ocp-installer-e2e-azure-4.5`. That one has some env vars to specify RELEASE_IMAGE_LATEST. It also has a configMap called prow-job-cluster-launch-installer-e2e.

And in other jobs we've enabled this, we've had to map e2e-metal-ipi to the workflow in the step registry, where does that happen for this job?

Comment (andfasano) [reply ci-operator/jobs/openshift/release/openshift-release-release-4.5-periodics.yaml]: This is what I've asked in the forum: all the jobs here are template based, while our is multi-stage based, so it's slightly different, and it was not clear the level of integration required.

Comment (smarterclayton) [reply ci-operator/jobs/openshift/release/openshift-release-release-4.5-periodics.yaml]: You need the template parameters, and if you aren't respecting them that would be an issue (it means we can't run upgrade jobs).

In general as the first adopter of steps there may be under documented steps, but we can find them out as we go.  You definitely need the BRANCH, RELEASE_IMAGE_LATEST, and RELEASE_IMAGE_INITIAL optional parameters.

**Comment by smarterclayton** [comment]
You need to declare the stub env vars like a release-*-origin-*installer job would.

File: ci-operator/jobs/openshift/release/openshift-release-release-4.5-periodics.yaml (hunk 1 above)
Comment (andfasano) [reply ci-operator/jobs/openshift/release/openshift-release-release-4.5-periodics.yaml]: We don't use RELEASE_IMAGE_INITIAL in the multi-stage test, maybe we can omit it. Is it ok to use `registry.svc.ci.openshift.org/ocp/release:4.5-ci`?

**Comment by andfasano** [comment]
> You need to declare the stub env vars like a release-*-origin-*installer job would.

I've added RELEASE_IMAGE_LATEST and BRANCH env vars, and also the additional args for the ci-operator, but it looks like the rehearse job is reporting the same error `error="ci-operator config file --.yaml was not found"`.
Not sure if other env vars are still required.

**Comment by stbenjam** [comment]
You're also missing `CONFIG_SPEC`. My attempt to add it didn't work: https://github.com/openshift/release/pull/7976

I tried this:


```yaml
          tests:
          - as: e2e-metal-ipi
            steps:
              cluster_profile: packet
              workflow: baremetalds-e2e
```

CI reports this:

```
2020/03/27 18:43:13 ci-operator version v20200327-eaf1a07
error: invalid configuration: tests[0]: non-literal test found in fully-resolved configuration
```

@smarterclayton Do you know how we make this work then? How do we let CI know that the step workflow it's supposed to run is baremetalds-e2e?

**Comment by andfasano** [comment]
> You're also missing `CONFIG_SPEC`. My attempt to add it didn't work: #7976
> 
> I tried this:
> 
> ```yaml
>           tests:
>           - as: e2e-metal-ipi
>             steps:
>               cluster_profile: packet
>               workflow: baremetalds-e2e
> ```
> 
> Things are now failing and a rehearse job gets created for the release informer.
> 
> But CI reports this:
> 
> ```
> 2020/03/27 18:43:13 ci-operator version v20200327-eaf1a07
> error: invalid configuration: tests[0]: non-literal test found in fully-resolved configuration
> ```
> 
> @smarterclayton Do you know how we make this work then? How do we let CI know that the step workflow it's supposed to run is baremetalds-e2e?

I've added `CONFIG_SPEC` with the same result as well, but it looks like a resolve step is missing (as per the error message). What's the difference in respect to the standard testing case where workflows get resolved, for example when the config is loaded from a specific folder like in https://github.com/openshift/release/blob/f9c4a09e2cdd83011cfc8e1d6d5bd8deca13194c/ci-operator/config/openshift/installer/openshift-installer-release-4.5.yaml#L198?

**Comment by petr-muller** [comment]
Oh wait, https://github.com/openshift/ci-tools/pull/624 haven't actually merged yet :/

**Comment by stbenjam** [comment]
Replaced by https://github.com/openshift/release/pull/8091

/close

**Pull request closed without merging**